### PR TITLE
build: Use .cjs file extension with CJS mocha configs

### PR DESCRIPTION
**Pull request by tylerbutler**
Renames all the `.mocharc.js` files to use the .cjs file extension. This will make it easier to add type: module to ESM-only packages.

I used these commands to do the bulk of the work:

```shell
fd --glob "**/.mocharc.js" --hidden --exec mv {} {.}.cjs
fd package.json --exec sd --fixed-strings '.mocharc.js' '.mocharc.cjs'
```

### CODE REVIEW

**Comment by Abe27342** [comment]
there are a few other references to `mocharc.js` in the repo which look like they need updating:

- mocha-test-setup README 
- e2e test packages have nested mocharcs with explicit extends blocks
- packages/dds/tree/.vscode/settings.json
OTW looks good

File: azure/packages/test/scenario-runner/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
 		"test:realsvc": "npm run test:realsvc:tinylicious",
 		"test:realsvc:azure": "cross-env FLUID_CLIENT=azure npm run test:realsvc:azure:run",
 		"test:realsvc:azure:run": "mocha --recursive \"dist/test/**/*.spec.*js\" --exit --timeout 10000",
-		"test:realsvc:run": "mocha dist/test --config src/test/.mocharc.js",
+		"test:realsvc:run": "mocha dist/test --config src/test/.mocharc.cjs",
 		"test:realsvc:tinylicious": "start-server-and-test start:tinylicious:test 7071 test:realsvc:azure:run",
```
Comment (jason-ha) [inline azure/packages/test/scenario-runner/package.json:37]: I don't see a file rename corresponding to this change.

**Review by jason-ha — CHANGES_REQUESTED** [review]
Nothing uses azure/packages/test/scenario-runner test:realsvc:run ?

**Comment by tylerbutler** [comment]
> Nothing uses azure/packages/test/scenario-runner test:realsvc:run ?

Doesn't look like it so I removed the script. My hunch is the project was copied from another, likely the e2e tests one, and unused scripts weren't removed at that time.

**Comment by tylerbutler** [comment]
> there are a few other references to `mocharc.js` in the repo which look like they need updating:
> 
> * mocha-test-setup README
> * e2e test packages have nested mocharcs with explicit extends blocks
> * packages/dds/tree/.vscode/settings.json
>   OTW looks good

Fixed in latest.

**Review by jason-ha — APPROVED** [review]

**Review by Abe27342 — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**